### PR TITLE
Amend configs to use Pydantic-style discriminator

### PR DESCRIPTION
**Pull request by DiamondJoseph**
- Remove Sinks that were used prior to isolated devices being supported

### CODE REVIEW

**Comment by DiamondJoseph** [comment]
Required https://github.com/dls-controls/tickit/pull/154

File: pyproject.toml [hunk 1]
```diff
@@ -12,7 +12,7 @@ classifiers = [
 ]
 description = "Devices for tickit, an event-based device simulation framework"
 dependencies = [
-    "tickit<0.2.3",
+    "tickit>=0.3,<0.4",
```
Comment (tpoliaw) [inline pyproject.toml:15]: Maybe leave as `>=0.3` and only limit it if there's a breaking change?

**Review by tpoliaw — COMMENTED** [review]
For all the `@pydantic.v1.dataclasses.dataclass` changes, would it make more sense to only change the import so we don't have to change the decorators as well? Does pydantic v2 have its own dataclasses as well or is that going to be a big enough change that we're going to have to change everything again anyway?

What's happening with all the eiger stuff being deleted? Is the new version a complete rewrite?

**Comment by DiamondJoseph** [comment]
@abbiemery @tpoliaw @callumforrester this is ready for review now that the eiger changes are in and everything is behaving.

**Review by tpoliaw — APPROVED** [review]

**Pull request merged**